openhands/runtime/runtime.py

```diff
@@ -106,7 +106,9 @@ def add_env_vars(self, env_vars: dict[str, str]) -> None:
                 f'Failed to add env vars [{env_vars}] to environment: {obs.content}'
             )
 
-    async def on_event(self, event: Event) -> None:
+    async def on_event(self, event_stream: EventStream, event: Event) -> None:
+        if event_stream is not self.event_stream:
+            raise ValueError('Event stream mismatch')
```

Comment:
Yeah, good question. In all of the other classes I think that there is no `self.event_stream` member. Arguably we don't need it in the runtime either, but I decided not to refactor this part for now because I thought it might be too much for one pull request.

Reply:
Isn't there? I think the classes that `add_event` have a member `self.event_stream` to do it. 🤔 For example, in AgentController, there is an eventstream and it uses it quite a lot, e.g. for delegates, for agent state changes: [event_stream](https://github.com/All-Hands-AI/OpenHands/blob/2a10ff13741dc47452240787b06def2463d38903/openhands/controller/agent_controller.py#L97)

AgentSession also has a reference to its `event_stream` [here](https://github.com/All-Hands-AI/OpenHands/blob/2a10ff13741dc47452240787b06def2463d38903/openhands/server/session/agent.py#L21).

Main and cli don't, but they're not a class... They have a variable and inner functions using it.